src/components/Button.js

```diff
@@ -152,7 +152,8 @@ class Button extends Component {
 
         // Setup and attach keypress handler for pressing the button with Enter key
         this.unsubscribe = KeyboardShortcut.subscribe(shortcutConfig.shortcutKey, (e) => {
-            if (!this.props.isFocused || this.props.isDisabled || this.props.isLoading || (e && e.target.nodeName === 'TEXTAREA')) {
+            if (!this.props.isFocused || this.props.isDisabled || this.props.isLoading
+                || (e && e.target.nodeName === 'TEXTAREA') || !this.props.pressOnEnter) {
                 return;
```

Comment:
Yeah definitely, it'd be better not to touch `Button`. I did this because `pressOnEnter` is enabled in `AttachmentModal` and that causes the PDF to be sent if the user uses the enter key to submit the password. So I must somehow disable `pressOnEnter` during the password challenge.
https://github.com/Expensify/App/blob/997746a2c30300f189875b731c0e7ec6cc0d37a6/src/components/AttachmentModal.js#L189-L195

So instead, I should take care of this inside `AttachmentModal` without touching `Button.js`. For example,
```
{this.props.onConfirm && (this.state.isSendOnEnterEnabled ? (
    <Button
        success
        style={[styles.buttonConfirm]}
        textStyles={[styles.buttonConfirmText]}
        text={this.props.translate('common.send')}
        onPress={this.submitAndClose}
        pressOnEnter
    />
) : (
    <Button
        success
        style={[styles.buttonConfirm]}
        textStyles={[styles.buttonConfirmText]}
        text={this.props.translate('common.send')}
        onPress={this.submitAndClose}
    />
))}
```

Another way to manage the press-on-enter issue is to add a separate send button to the password form. So if a PDF attachment requires a password I do the following:
1. Hide/deactivate the send button in `AttachmentModal`
2. Render the password form with a text input, a `Confirm` button, **and** a `Send` button. This send button calls the send method in `AttachmentModal` when clicked.

Reply:
Unfortunately that didn't work. I just pushed a different potential solution. It still requires a change to `Button.js` but all I'm doing is adding the event to the `onPress` call:
```
this.unsubscribe = KeyboardShortcut.subscribe(shortcutConfig.shortcutKey, (e) => {
    if (!this.props.isFocused || this.props.isDisabled || this.props.isLoading || (e && e.target.nodeName === 'TEXTAREA')) {
        return;
    }
    this.props.onPress(e);
}, shortcutConfig.descriptionKey, shortcutConfig.modifiers, true, false, this.props.enterKeyEventListenerPriority, false);
```

If this isn't permissible then I think we should just hide/disable the send button in `AttachmentModal` and add a send button to the password form.